src/odemis/driver/test/piezomotor_test.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+Created on 18 Mar 2020
+
+@author: Philip Winkler
+
+Copyright © 2020, Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+from __future__ import division, print_function
+
+import logging
+import os
+import unittest
+from odemis.model import Axis
+from odemis.driver.piezomotor import PMD401Bus
+from collections import deque
+
+logging.getLogger().setLevel(logging.DEBUG)
+
+# Export TEST_NOHW=1 to force using only the simulator and skipping test cases
+# needing real hardware
+TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)  # Default to Hw testing
+
+class TestPMD401(unittest.TestCase):
+    """
+    Test the PMD401 class.
+    """
+
+    def setUp(self):
+        if TEST_NOHW:
+            port = "/dev/fake"
+        else:
+            port = "/dev/ttyUSB*"
+
+        axes = {"x": {"axis_number": 1,
+                      "mode": 1,
```

Comment:
I don't think "mode" does anything. It could be good to raise an error at instantiation if this happens (to do so easily, use `.pop()` on the dict, and raise an error if it's not empty at the end of the parsing).